Sources/BraintreePayPal/BTPayPalDriver.m

```diff
@@ -124,6 +122,14 @@ - (void)requestPayPalCheckout:(BTPayPalRequest *)request
         return;
     }
 
+    if (!([request isKindOfClass:BTPayPalCheckoutRequest.class] || [request isKindOfClass:BTPayPalVaultRequest.class])) {
+        NSError *error = [NSError errorWithDomain:BTPayPalDriverErrorDomain
+                                             code:BTPayPalDriverErrorTypeIntegration
+                                         userInfo:@{NSLocalizedDescriptionKey: @"BTPayPalDriver failed because request is not of type BTPayPalCheckoutRequest or BTPayPalVaultRequest."}];
+        completionBlock(nil, error);
+        return;
+    }
```

Comment:
Is there a way to hide the constructor for `BTPayPalRequest` from the merchant to prevent this from being possible?

Reply:
I'm not sure... I know that you can mark initializers as "unavailable" in Objective-C, but I think that might prevent the SDK from using that initializer too. 🤔

Reply:
That makes sense yeah I think it might be possible in Swift but not Obj-C.